src/lib/list/selection-list.ts

```diff
@@ -267,36 +240,23 @@ export class MdSelectionList extends _MdSelectionListMixinBase
     });
   }
 
-  /** Map all the options' destroy event subscriptions and merge them into one stream. */
-  private _onDestroySubscription(): Subscription {
-    return RxChain.from(this.options.changes)
-      .call(startWith, this.options)
-      .call(switchMap, (options: MdListOption[]) => {
-        return merge(...options.map(option => option.destroyed));
-      }).subscribe((e: MdSelectionListOptionEvent) => {
-        let optionIndex: number = this.options.toArray().indexOf(e.option);
-        if (e.option._hasFocus) {
-          // Check whether the option is the last item
-          if (optionIndex < this.options.length - 1) {
-            this._keyManager.setActiveItem(optionIndex);
-          } else if (optionIndex - 1 >= 0) {
-            this._keyManager.setActiveItem(optionIndex - 1);
-          }
-        }
-        e.option.destroyed.unsubscribe();
-      });
+  /** Sets the focused option of the selection-list. */
+  _setFocusedOption(option: MdListOption) {
+    this._keyManager.updateActiveItemIndex(this._getIndexFromOption(option));
   }
 
-  /** Map all the options' onFocus event subscriptions and merge them into one stream. */
-  private _onFocusSubscription(): Subscription {
-    return RxChain.from(this.options.changes)
-      .call(startWith, this.options)
-      .call(switchMap, (options: MdListOption[]) => {
-        return merge(...options.map(option => option.onFocus));
-      }).subscribe((e: MdSelectionListOptionEvent) => {
-      let optionIndex: number = this.options.toArray().indexOf(e.option);
-      this._keyManager.updateActiveItemIndex(optionIndex);
-    });
+  /** Removes an option from the selection list and updates the active item. */
+  _removeOptionFromList(option: MdListOption) {
+    if (option._hasFocus) {
+      const optionIndex = this._getIndexFromOption(option);
+
+      // Check whether the option is the last item
+      if (optionIndex - 1 >= 0) {
+        this._keyManager.setPreviousItemActive();
+      } else if (optionIndex < this.options.length - 1) {
```

Comment:
I don't see any case, aside from `optionIndex === 0`, where it would get into this block. You can simplify it to check for `optionIndex === 0` instead.

Reply:
Done. Addressed your feedback.